src/controllers/blog-insight.ts

```diff
@@ -0,0 +1,130 @@
+import { Request, Response, NextFunction } from "express";
+import { UUID } from "crypto";
+import { BlogInsightRepository } from "../infrastructure/blog-insight-impl";
+import { BlogInsightService } from "../services/blog-insight";
+import { CreateBlogInsightSchema, UpdateBlogInsightSchema } from "../schemas/blog-insight";
+import { IdSchema } from "../schemas/id";
+import { UnprocessableEntityException } from "../exceptions/validation";
+import { UnauthorizedException } from "../exceptions/unauthorized";
+import { BadRequestException } from "../exceptions/bad-request";
+import { ErrorCode } from "../exceptions/root";
+import { CreateBlogInsightDto, UpdateBlogInsightDto } from "../dtos/blog-insight";
+import { CreateBlogInsight, UpdateBlogInsight } from "../interfaces/blog-insight";
+
+const blogInsightRepository = new BlogInsightRepository();
+export const blogInsightService = new BlogInsightService(blogInsightRepository);
+
+export const createBlogInsight = async (req: Request, res: Response, next: NextFunction) => {
+
+    const blogIdParam = req.params.blogId;
+    const parsedBlogId = IdSchema.safeParse(blogIdParam);
```

Comment:
always try to use less variable prefrably don't use variable if not use more than 1 time . Like you can write here easily

```suggestion
    const parsedBlogId = IdSchema.safeParse(req.params.blogId);
```